src/api/cubrid_log.c

```diff
@@ -599,21 +608,60 @@ cubrid_log_connect_server_internal (char *host, int port, char *dbname)
 
   reason = ntohl (*(int *) recv_data);
 
-  if (reason != SERVER_CONNECTED)
+  if (recv_data != NULL)
     {
-      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT, NULL);
+      free_and_init (recv_data);
     }
 
-  if (recv_data != NULL && recv_data != (char *) &reason)
+#if defined (WINDOWS)
+  if (reason == SERVER_CONNECTED_NEW)
     {
-      free_and_init (recv_data);
+      if (css_receive_data (g_conn_entry, rid, &recv_data, &recv_data_size, g_connection_timeout * 1000) != NO_ERRORS)
+	{
+	  CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT,
+				     "Failed to receive the server port id from the master.\n");
+	}
+
+      if (recv_data != NULL)
+	{
+	  assert (recv_data_size == sizeof (int));
+
+	  int port_id = ntohl (*((int *) recv_data));
+
+	  css_close_conn (g_conn_entry);
+
+	  g_conn_entry = css_server_connect_part_two (host, g_conn_entry, port_id, &rid);
+	  if (g_conn_entry == NULL)
+	    {
+	      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT,
+					 "Failed to connect to the server with new port id (%d)\n", port_id);
+	    }
+
+	  free_and_init (recv_data);
+	}
+      else
+	{
```

Comment:
@wrlawodms 
The reason I didn't do css_close_conn() here is that 
when an error occurs, the label (`cubrid_log_error:`) cleans up the connection through css_free_conn().

Reply:
Oh, Thanks. I will fix it. 
it should check if `g_conn_entry` is NULL before clear the `queue_entry`.